src/math/dd/dd_pdd.cpp

```diff
@@ -743,10 +745,122 @@ namespace dd {
                 rest = p;
             }
         }
-        e.m_lc = lc.root;
-        e.m_rest = rest.root;
+        if (cache_generation != m_factor_cache_generation) {
+            // Cache was reset while factoring (due to GC),
+            // which means the old entry has been removed and we need to insert it again.
+            auto* et = m_factor_cache.insert_if_not_there2({p.root, v, degree});
+            e = &et->get_data();
+        }
+        e->m_lc = lc.root;
+        e->m_rest = rest.root;
+    }
+
+    template <class Fn>
+    pdd pdd_manager::map_coefficients(pdd const& p, Fn f)
+    {
```

Comment:
convention: { on previous line.

Reply:
this could also be memoized with the op-cache. 
map-coefficients isn't nested with different Fn so the same op-code can be used.

Reply:
op-cache use has to be very careful: it persists across calls and the calls could use different divisors.
So the divisor itself needs to be on the cache object. This is possible as op-cache takes 2 arguments + op.
The constant node corresponding to pow2j would be one of the arguments.

Reply:
And map_coefficients needs to be less generic as well (not allow arbitrary functions like now).

Reply:
of course, it would be a different function, similar to minus and minus_rec